### PR TITLE
vdev_id: extension for new scsi topology.

### LINKED ISSUE #6592: Extension to vdev_id script
I have a server which presents scsi disks rather than sas ones.  It is therefore incompatible with the vdev_id script.

I have made some local modifications to this script to support what I am calling "scsi" topology.  It works for me, but I am very aware that testing this sort of thing thoroughly requires access to a variety of hardware which I don't have.

Are you interested in a pull request with this extension?

### PR DESCRIPTION
**Pull request by tesujimath**
On systems with SCSI rather than SAS disk topology, this change enables
the vdev_id script to match against the block device path, and therefore
create a vdev alias in /dev/disk/by-vdev.

Resolves #6592.

Signed-off-by: Simon Guest <simon.guest@tesujimath.org>

### Description
An additional feature allows to set the bay number of the first disk, to provide an offset from the SCSI default of 0, to match physical bay numbers, which may be numbered from 1.

### Motivation and Context
Without this enhancement, servers with such a SCSI disk topology are unable to have their zpools configured with the very convenient vdev_id disk names.

For now, this has been implemented as a new scsi_handler function in the vdev_id script, to avoid disturbing the existing functionality.  There is some commonality between the new scsi_handler and the existing sas_handler, but it is not clear to me how best to address that.

Resolves #6592.

### How Has This Been Tested?
Installed script and matching config on a server with this topology.  Observed that it produced the correct symlinks in /dev/disk/by-vdev.
(This is a manual test;  unclear to me how to automate such a thing.)

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by behlendorf** [comment]
@tonyhutter when you get a chance can you verify this doesn't cause any unexpected problems on our systems.  It shouldn't.

**Comment by tonyhutter** [comment]
@behlendorf the new vdev_id works for me on our test system.  I was able to to import our pool without issue.

File: etc/zfs/vdev_id.conf.scsi.example [hunk 1]
```diff
@@ -0,0 +1,9 @@
+multipath     no
+topology      scsi
+phys_per_port 1
+# Usually scsi disks are numbered from 0, but this can be offset, to
+# match the physical bay numbers, as follows:
+first_bay_number 1
+
+#       PCI_ID  HBA PORT  CHANNEL NAME
+channel 0c:00.0 0         Y
```
Comment (behlendorf) [inline etc/zfs/vdev_id.conf.scsi.example]: You need to add this file to the `Makefile.am`.

**Comment by tesujimath** [comment]
@behlendorf Thanks!

**Pull request merged**